app/main/views/agreements.py

```diff
@@ -28,7 +28,9 @@ def _get_ordered_supplier_frameworks(framework_slug, status=None):
         **({"statuses": status} if status else {})
     )['supplierFrameworks']
 
-    # API now returns SupplierFrameworks by agreementReturnedAt ascending (oldest first)
+    # Timestamps in our API responses are formatted using dmutils DATETIME_FORMAT which sorts in lexicographical order
+    # i.e. "%Y-%m-%dT%H:%M:%S.%fZ" - we want the newest things at the bottom of the list so this ordering is correct
+    supplier_frameworks = sorted(supplier_frameworks, key=lambda k: k.get("agreementReturnedAt", "2000-01-01"))
```

Comment:
Maybe a short comment to say that "2000-01-01" is an arbitrary date to enable ordering for supplier frameworks that are not returned. If we hadn't already discussed this before I don't think I would have got this particularly quickly.